global_helpers/panther_oss_helpers.py

```diff
@@ -1,14 +1,76 @@
 """Utility functions provided to policies and rules during execution."""
+import re
 import os
 import time
-from typing import Any, Dict, Sequence, Set, Union
+from datetime import datetime
+from typing import Any, Dict, Sequence, Set, Union, Optional
 
 import boto3
 
 _RESOURCE_TABLE = None  # boto3.Table resource, lazily constructed
 FIPS_ENABLED = os.getenv("ENABLE_FIPS", "").lower() == "true"
 FIPS_SUFFIX = "-fips." + os.getenv("AWS_REGION", "") + ".amazonaws.com"
 
+# Auto Time Resolution Parameters
+EPOCH_REGEX = r"[^0-9\-]+([-]?[0-9]{,12})"
+TIME_FORMATS = {
+    "%Y-%m-%dT%H:%M:%SZ",  # AWS Timestamp
+    "%Y-%m-%dT%H:%M:%S.%fZ",  # Panther Timestamp
```

Comment:
So I'm not 100% sure _why_ this is causing an issue, but python iterates sets in an arbitrary order between runs (see screenshot below, I loaded this set in two different interpreters exactly as you have it here and then iterated them). This might explain why _sometimes_ certain tests fail and certain tests don't.

<img width="1049" alt="Screen Shot 2021-04-16 at 9 40 06 AM" src="https://user-images.githubusercontent.com/49166439/115056455-bd69a600-9e97-11eb-9f8e-3bbd098e37ce.png">

The only thing I can think is that some of the time stamp formats below are matching the timestamp getting put in, but are not returning a value close enough to what the test is expecting. Some of the unit tests are very granular, on the order of seconds.

So my first recommendation: change this set to a list. We'r'e not doing any membership checking, so there isn't a performance concern really. Even if there were, I think reliability is more important in this case.

Secondly though, we should figure out which timestamp formats are causing some tests to fail when other formats cause it to pass.